pkg/types/types.go

```diff
@@ -30,19 +33,63 @@ const (
 	KubeEndPoint = "kubelet.sock"
 )
 
-// ResourceConfig contains cofiguration paremeters for a resource pool
+// DeviceType is custom type to define supported device types
+type DeviceType string
+
+const (
+	// NetDeviceType is DeviceType for network class devices
+	NetDeviceType DeviceType = "netDevice"
+	// AcceleratorType is DeviceType for accelerator class devices
+	AcceleratorType DeviceType = "accelerator"
+)
+
+// SupportedDevices is map of 'device identifier as string' to 'device class hexcode as int'
+/*
+Network controller subclasses. ref: https://pci-ids.ucw.cz/read/PD/02
+00	Ethernet controller
+01	Token ring network controller
+02	FDDI network controller
+03	ATM network controller
+04	ISDN controller
+05	WorldFip controller
+06	PICMG controller
+07	Infiniband controller
+08	Fabric controller
+80	Network controller
+*/
+var SupportedDevices = map[DeviceType]int{
```

Comment:
I dont really like the coupling here between the device PCI class code and the device type.

what if in the future, we would like to add a device type which would handle the same PCI class but different attributes ?
Or a single device type (device Provider) handling several PCI classes ?

Reply:
Well for now it make sense to use the device class code to differentiate different device types. Same class different attributes - the we that device specific selectors to handle additional logic etc.
I don't know what better way to determine different deviceType. You have any recommendation?